### DISCUSSION TITLE
HELP: additional attributes based on HttpServletRequest headers

### DISCUSSION THREAD

**Discussion by bgong-mdsol** [open-telemetry/opentelemetry-java-instrumentation#4224] [Q&A] [answered]
I have a instrumentation to add the span attributes based on HttpServletRequest headers. It works fine with opentelemetry-java & opentelemetry-java-agent 1.5.x. When try upgrade to 1.6.0, the additional attributes are not available :(

Here is the helper class to add additional attributes:
```
import static io.opentelemetry.instrumentation.servlet.v3_0.Servlet3HttpServerTracer.tracer;
import io.opentelemetry.api.common.Attributes;
import io.opentelemetry.api.common.AttributesBuilder;
import io.opentelemetry.api.trace.Span;
import io.opentelemetry.context.Context;
import javax.servlet.http.HttpServletRequest;

public class HttpAttributesHelper {
  ...
  public static void updateSpanAttributes(HttpServletRequest httpRequest) {
    Attributes attributes = getAdditionalSpanAttributes(httpRequest); // Additional Attributes for span 
    if (!attributes.isEmpty()) {
      Context attachedContext = tracer().getServerContext(httpRequest);
      Span span = Span.fromContext(attachedContext);
      span.setAllAttributes(attributes);
      span.makeCurrent();
    }
  }
  ...
}

```
Since `io.opentelemetry.instrumentation.servlet.v3_0.Servlet3HttpServerTracer` is deprecated in opentelemetry-java & opentelemetry-java-agent 1.6.0, I have changed `io.opentelemetry.javaagent.instrumentation.servlet.v3_0.Servlet3Singletons.helper` to replace `tracer()`, the additional attributes shows as expected in junit test, but I can't find the attributes when running Smoke test on sprint boot platform. 

I am not sure what I should update to use opentelemetry-java & opentelemetry-java-agent 1.6.0. Can someone help? Thanks in advance.

**Comment by iNikem** [comment, marked as answer]
Can you try the following code?

```
public class HttpAttributesHelper {
  ...
  public static void updateSpanAttributes(HttpServletRequest httpRequest) {
    Attributes attributes = getAdditionalSpanAttributes(httpRequest); // Additional Attributes for span 
    if (!attributes.isEmpty()) {
      Span span = Span.current();
      span.setAllAttributes(attributes);
    }
  }
  ...
}
```

**Reply by bgong-mdsol** [reply]
It works. Thanks a lot @iNikem
